### ISSUE TITLE
prod() backward pass is unimplemented — todo!() panic

### ISSUE THREAD

**Issue by zk4x** [zk4x/zyx#3]
### Description

Calling `prod()` on a tensor and then `.backward()` panics with `todo!()` in `autograd.rs:534`.

The gradient for `Reduce(BOp::Mul)` is not implemented:

```rust
BOp::Mul => todo!(),
```

### Reproduction

```rust
let x = Tensor::from([[1.0, 2.0], [3.0, 4.0]]);
let y = x.prod([1])?;
let loss = y.sum([-1])?;
let grads = loss.backward([&x])?;  // panic
```

### Expected Behavior

Backward pass computes correct gradients instead of panicking.
